arviz/plots/backends/matplotlib/ecdfplot.py

```diff
@@ -0,0 +1,189 @@
+            )
+        else:
+            x = np.linspace(values[0], values[-1], granularity)
+            z = distribution(x) if distribution else compute_ecdf(values2, x) / len(values2)
+
+        gamma = get_gamma(n, z, granularity, num_trials, alpha)
+        lower, higher = get_lims(gamma, n, z)
+
+        x_coord, y_coord = [], []
+        if not difference:
+            for _, x_i in enumerate(x):
+                f_x_i = compute_ecdf(values, x_i) / n
+                x_coord.append(x_i)
+                y_coord.append(f_x_i)
+
+            plt.step(x_coord, y_coord, where="post")
+
+            if ecdf_fill:
+                plt.fill_between(x, lower / n, higher / n, color="C0", alpha=0.2)
+            else:
+                plt.plot(x, lower / n, x, higher / n, color="C0", alpha=0.2)
+        else:
+            for _, x_i in enumerate(x):
+                f_x_i = compute_ecdf(values, x_i) / n - (
+                    x_i
+                    if pit
+                    else distribution(x_i)
+                    if distribution
+                    else compute_ecdf(values2, x_i) / len(values2)
+                )
+                x_coord.append(x_i)
+                y_coord.append(f_x_i)
+
+            plt.step(x_coord, y_coord, where="post")
+            if pit:
+                lower = lower / n - x
+                higher = higher / n - x
+            else:
+                lower = lower / n - (
+                    distribution(x) if distribution else compute_ecdf(values2, x) / len(values2)
+                )
+                higher = higher / n - (
+                    distribution(x) if distribution else compute_ecdf(values2, x) / len(values2)
+                )
+
+            if ecdf_fill:
+                plt.fill_between(x, lower, higher, color="C0", alpha=0.2)
+            else:
+                plt.plot(x, lower, x, higher, color="C0", alpha=0.2)
+
+    else:
+        if pit:
+            x = np.linspace(1 / granularity, 1, granularity)
+            values = distribution(values)
+        else:
+            x = np.linspace(values[0], values[-1], granularity)
+
+        x_coord, y_coord = [], []
+        if not difference:
+            for _, x_i in enumerate(x):
+                f_x_i = compute_ecdf(values, x_i) / n
+                x_coord.append(x_i)
+                y_coord.append(f_x_i)
+
+            plt.step(x_coord, y_coord, where="post")
+        else:
+            for _, x_i in enumerate(x):
+                f_x_i = compute_ecdf(values, x_i) / n - (
+                    x_i
+                    if pit
+                    else distribution(x_i)
+                    if distribution
+                    else compute_ecdf(values2, x_i) / len(values2)
+                )
+                x_coord.append(x_i)
+                y_coord.append(f_x_i)
+
+            plt.step(x_coord, y_coord, where="post")
+
+    if backend_show(show):
+        plt.show()
+
+    return ax
+
+
+def compute_ecdf(sample, z):
```

Comment:
Technically this is not computing the ECDF; it's computing the rank of `z` in `sample` (because it doesn't divide by the denominator). Should the name reflect that?

Reply:
Changed, I have added the it in the denominator rather than explicitly doing it outside